### PR TITLE
fix: variables values interpolation

### PR DESCRIPTION
**Pull request by denis256**
## Description

* added variable values replacement
* added basic integration test for handling replacements
* updated lint go settings to reduce process killing

Fix: #3368

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] I authored this code entirely myself
- [x] I am submitting code based on open source software (e.g. MIT, MPL-2.0, Apache)]
- [x] I am adding or upgrading a dependency or adapted code and confirm it has a compatible open source license
- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

## Summary by CodeRabbit

* **Bug Fixes**
  * Preserve Terraform-style interpolation patterns (e.g., ${...}) as literal strings inside maps, lists, and nested inputs so they are passed to Terraform unchanged.

* **Tests**
  * Added unit and integration tests covering interpolation handling, edge cases, and nesting-depth behavior.

* **Chores**
  * Increased memory limits for Linux lint runs in CI.

### CODE REVIEW

File: internal/util/jsons.go [hunk 1]
```diff
@@ -2,23 +2,136 @@ package util
...
 		if err != nil {
 			return "", err
 		}
 
 		return string(envVarValue), nil
 	}
 }
+
+// escapeInterpolationPatternsInValue recursively walks a value tree and escapes
+// HCL interpolation patterns (${...}) in string values to prevent Terraform from
```
Comment (yhakbar) [inline internal/util/jsons.go]:
```suggestion
// HCL interpolation patterns (${...}) in string values to prevent OpenTofu/Terraform from
```

File: internal/util/jsons.go [hunk 2]
```diff
@@ -2,23 +2,136 @@ package util
...
+// treating them as variable references when parsing complex type TF_VAR_* env vars.
+//
+// This unconditionally escapes ${...} in all string values within complex types.
+// This is intentional: Terraform's HCL parser would error on unescaped ${...} in
```
Comment (yhakbar) [inline internal/util/jsons.go]:
```suggestion
// This is intentional: OpenTofu/Terraform's HCL parser would error on unescaped ${...} in
```

**Review by yhakbar — DISMISSED** [review]
